hoot-test/src/main/cpp/hoot/test/main.cpp

```diff
@@ -564,31 +562,38 @@ int main(int argc, char *argv[])
         populateTests(GLACIAL_ONLY, vAllTests, printDiff);
       }
 
+      vector<CppUnit::Test*> vTests;
+      getTestVector(vAllTests, vTests);
       bool filtered = false;
+
       for (int i = 0; i < args.size(); i++)
       {
         if (args[i].startsWith("--exclude="))
         {
+          if (vTestsToRun.size() > 0)
+          {
+            //  On the second (or more) time around exclude from the excluded list
+            vTests.swap(vTestsToRun);
+            vTestsToRun.clear();
+          }
```

Comment:
This is where the magic happens, for each exclusion start with the already filtered list and push them onto an empty list.